xmtp_mls/src/groups/mod.rs

```diff
@@ -5769,4 +5769,175 @@ pub(crate) mod tests {
             .await;
         assert!(result.is_err());
     }
+
+    #[xmtp_common::test(flavor = "multi_thread")]
+    async fn can_stream_out_of_order_without_forking() {
```

Comment:
Verified test would fork the group before the fix see test PR: https://github.com/xmtp/libxmtp/pull/1862

<img width="784" alt="image" src="https://github.com/user-attachments/assets/aa60f49f-9d1e-4e0b-8e69-b73a8c40e734" />